app/controllers/shelters_controller.rb

```diff
@@ -34,7 +46,7 @@ def destroy
   private
 
   def shelter_params
-    params.permit(:name, :address, :city, :state, :zip)
+    params.permit(:name, :address, :city, :state, :zip).to_h
```

Comment:
Change params to hash to use in model method to show keys of values that are empty strings.